app.py

```diff
@@ -520,9 +520,17 @@ def prepare_generic_launch(self, engine_name, context, app_path, app_args):
             self.log_exception("Error executing engine bootstrap script.")
             raise TankError("Error executing bootstrap script. Please see log for details.")
         finally:
-            # remove bootstrap from sys.path
+            # Remove bootstrap from sys.path
             sys.path.pop(0)
 
+            # We also need to unload the bootstrap module so that any
+            # subsequent launches that import a different bootstrap
+            # will succeed.
+            try:
+                del sys.modules["bootstrap"]
+            except Exception:
+                pass
```

Comment:
Perhaps log the deletion if it happens in a `finally`?

Also, any reason for `try`/`except` over:

```python
    if "bootstrap" in sys.modules:
        self.log_debug("Removing 'bootstrap' from sys.modules.")
        del sys.modules["bootstrap"]
```